### PR TITLE
Support Keycloak Authorization for OIDC web-app applications

### LINKED ISSUE #17193: Support Keycloak Authorization for OIDC 'web-app' applications
## Description

Pedro @pedroigor proposed it awhile back, it was not quite obvious how it would work with `web-app` where ID token is a primary token. But now that the access token can be configured as a source of roles we can support it for `web-app`.

## Implementation ideas

It should work straight away - we just need to relax a check a bit - KC Authorization can not be supported only if it is `web-app` and the source of roles is not set to `access-token`

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #17193

Hi Pedro @pedroigor it is nearly ready, but sadly I can't get the test passing, getting `403` for `alice` - she is `superuser` so I expect `200` given a new `Superuser` JsPolicy - can you spot the problem with the test :-) ?

### CODE REVIEW

File: integration-tests/keycloak-authorization/src/main/resources/application.properties [hunk 1]
```diff
@@ -59,22 +59,24 @@ quarkus.keycloak.policy-enforcer.paths.10.enforcement-mode=ENFORCING
...
 
+# Web App Tenant
+quarkus.oidc.webapp-tenant.auth-server-url=${keycloak.url}/realms/quarkus
+quarkus.oidc.webapp-tenant.client-id=quarkus-app
+quarkus.oidc.webapp-tenant.credentials.secret=secret
+quarkus.oidc.webapp-tenant.application-type=web-app
+quarkus.oidc.webapp-tenant.roles.source=accesstoken
+
+quarkus.keycloak.webapp-tenant.policy-enforcer.paths.1.name=Permission Resource WebApp
+quarkus.keycloak.webapp-tenant.policy-enforcer.paths.1.path=/api-permission-webapp
```
Comment (pedroigor) [inline integration-tests/keycloak-authorization/src/main/resources/application.properties:79]: Did you define a `/api-permission-webapp` endpoint?

Comment (sberyozkin) [reply integration-tests/keycloak-authorization/src/main/resources/application.properties:79]: @pedroigor My fault, it was there but I just forgot to add a test resource :-). But yes, it is still failing

Comment (sberyozkin) [reply integration-tests/keycloak-authorization/src/main/resources/application.properties:79]: @pedroigor I'll debug a bit more

Comment (sberyozkin) [reply integration-tests/keycloak-authorization/src/main/resources/application.properties:79]: @pedroigor I fixed it, the endpoint protected by `web-app` had to have `@Authenticated` added - without it the initial HTTP request without a token was considered anonymous

**Review by pedroigor — APPROVED** [review]

**Pull request merged**